### ISSUE TITLE
Repeated libE calls fail

### ISSUE THREAD

**Issue by jmlarson1** [Libensemble/libensemble#373]
@jlnav Can you figure out the best way to address the failure of
`libensemble/tests/regression_tests/test_evaluate_existing_sample.py`
on the `debugging/repeated_calls_fail` branch? 

- The same test is re-run but the second run fails. 
- If you run the test once by changing the `2` on line `25` of the test to `1`, the test completes.
- I think this has to do with the fact that `persis_info` is not reset correctly in the second call (although `persis_info` is not a `libE` argument). 
- Perhaps we need to clear `persis_info` (but adding `del persis_info` after the call doesn't help)

**Comment by jlnav** [comment]
For the above case, I think this is addressed by adding  ``persis_info[‘next_to_give’] = 0`` after the ``libE()`` call.

A puzzling observation: if ``print(persis_info)`` is added as the first statement in ``libE()``, the first output is ``{}`` since ``persis_info`` is instantiated as such in the ``def libE(..., persis_info = {}, ...)`` definition. But the next ``libE()`` call prints ``{'next_to_give': 1001}`` even though  ``persis_info`` is never passed as a parameter and thus (I assumed) should be instantiated as in the function definition.

We want ``libE()`` to instantiate ``persis_info={}`` every time if it's not passed as a keyword argument. That doesn't appear guaranteed right now.

**Comment by jmlarson1** [comment]
This is quite confusing to me. 

If `persis_info` isn't passed to `libE`, how is it getting that old value. 

Maybe @shuds13 can clarify what's going on.

**Comment by jmlarson1** [comment]
Running a script containing
````
def test(a,b={}):
    print(b)
    b[1] = a

test(1)
test(2)
````
results in the output:
````
{}
{1: 1}
````

That's unexpected to me.

**Comment by jmlarson1** [comment]
We should make the changes suggested here and use `None` instead of passing the empty dictionary.

http://effbot.org/zone/default-values.htm

**Comment by jlnav** [comment]
Closed by #378

**Comment by jmlarson1** [comment]
@jlnav It seems that repeated calls still fail at times (but for a different reason). 

See the example in `libensemble/libensemble/tests/regression_tests/test_persistent_aposmm_periodic.py`
in the branch `feature/periodic_aposmm_test`

`persis_info` is correctly being initialized, but the second call fails.

**Comment by jlnav** [comment]
> @jlnav It seems that repeated calls still fail at times (but for a different reason).
> 
> See the example in `libensemble/libensemble/tests/regression_tests/test_persistent_aposmm_periodic.py`
> in the branch `feature/periodic_aposmm_test`
> 
> `persis_info` is correctly being initialized, but the second call fails.

I can't reproduce the issue on my laptop when running with multiprocessing. When running with MPI I get the following on the second call:

``AssertionError: Aborting: Unknown calculation status received. Received status: 3``

Is this what you're seeing?

**Comment by jmlarson1** [comment]
Yes. That is exactly what I'm seeing.

**Comment by jlnav** [comment]
Tentatively closed by #392 and #393 ?
